pubsub/snapshot.go

```diff
@@ -42,11 +43,32 @@ func (s *Snapshot) ID() string {
 	return s.name[slash+1:]
 }
 
+// SetLabels replaces the current set of labels completely with the new set.
+func (s *Snapshot) SetLabels(ctx context.Context, label map[string]string) (*SnapshotConfig, error) {
+	var fields []string
+	fields = append(fields, "labels")
+	sc, err := s.c.subc.UpdateSnapshot(ctx, &pb.UpdateSnapshotRequest{
+		Snapshot: &pb.Snapshot{
+			Name:   s.name,
+			Labels: label,
+		},
+		UpdateMask: &fmpb.FieldMask{
+			Paths: fields,
```

Comment:
you don't appear to manipulate the fields in the mask, maybe just declare the paths using a literal ex:

`Paths: []string{"labels"},`